errors_113.go

```diff
@@ -0,0 +1,70 @@
+// This file implements the helper interfaces for the errors package introduced in Go 1.13
+// +build go1.13
+
+package errorx
+
+import "errors"
+
+// As implements the helper interface for errors.As.
+// If v is a pointer to a pointer to Error, *v will be set to this error.
+// Otherwise, it traverses the error chain until it reaches a non-*Error error and calls errors.As on that error and the
+// target.
+func (e *Error) As(v interface{}) bool {
+	if target, ok := v.(**Error); ok {
+		*target = e
+		return true
+	}
+
+	cause := e
+	for cause != nil {
+		next := cause.Cause()
+		cause = Cast(next)
+		if cause == nil && next != nil {
```

Comment:
If we check the result of `Cast(next)` for nil directly, it would clearly show that we only use it for "errorx type" check and nothing else.

Reply:
But we have to set it to `cause` in order to continue through the loop, so we would end up with just:

```
if Cast(next) == nil && next != nil {
    ...
}
cause = Cast(next)
```

Reply:
My mistake.